default.vcl.tmpl

```diff
@@ -226,11 +239,21 @@ sub vcl_deliver {
   # bans are in use. See bug 1492 for details. So take hits with a grain of salt
   set resp.http.X-Cache-Hits = obj.hits;
 
+  # Disable ETag for ESI pages as the ETag of subrequests doesn't propagate to the parent
+  # Pages served would always rely on the cache of the main page and therefore
+  # never refresh, even if the subrequest is updated
+  if (beresp.http.Surrogate-Control ~ "ESI/1.0") {
+    unset resp.http.ETag;
+    unset resp.http.Last-Modified;
+  }
+
   # Remove some headers
   unset resp.http.Server;
   unset resp.http.X-Varnish;
   unset resp.http.Via;
   unset resp.http.X-Generator;
+  unset resp.http.Surrogate-Control;
+  unset resp.http.X-Varnish-Grace;
```

Comment:
We could check whether we can easily move that into `Surrogate-Control`.
E.g. `Surrogate-Control: max-age=604800, stale-while-revalidate=86400`
and `Cache-Control: max-age=604800, stale-while-revalidate=86400`